lib/pengine/tests/status/pe_find_node_id_test.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 the Pacemaker project contributors
+ *
+ * The version control history for this file may have further details.
+ *
+ * This source code is licensed under the GNU Lesser General Public License
+ * version 2.1 or later (LGPLv2.1+) WITHOUT ANY WARRANTY.
+ */
+
+#include <crm_internal.h>
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <setjmp.h>
+#include <cmocka.h>
+
+#include <crm/pengine/internal.h>
+
+static void
+empty_list(void **state) {
+    assert_null(pe_find_node_id(NULL, NULL));
+    assert_null(pe_find_node_id(NULL, "id1"));
+}
+
+static void
+non_null_list(void **state) {
+    GList *nodes = NULL;
+
+    pe_node_t *a = calloc(1, sizeof(pe_node_t));
+    pe_node_t *b = calloc(1, sizeof(pe_node_t));
+
+    a->details = calloc(1, sizeof(struct pe_node_shared_s));
+    a->details->id = "id1";
+    b->details = calloc(1, sizeof(struct pe_node_shared_s));
+    b->details->id = "id2";
+
+    nodes = g_list_append(nodes, a);
+    nodes = g_list_append(nodes, b);
+
+    assert_ptr_equal(a, pe_find_node_id(nodes, "id1"));
+    assert_ptr_equal(b, pe_find_node_id(nodes, "ID2"));
+    assert_null(pe_find_node_id(nodes, "xyz"));
```

Comment:
It's also worth testing that substrings don't match, e.g. id10 or nodeid1

Reply:
Fixed.